gcs/tests/integration/requirer-charm/src/events/requirer.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python3
+# Copyright 2025
+# SPDX-License-Identifier: Apache-2.0
+import json
+import logging
+from typing import Any, Dict, Mapping, Optional
+
+import ops
+from charms.data_platform_libs.v0.object_storage import (
+    StorageRequires,
+)
+from ops.charm import CharmBase
+from ops.framework import Object
+from ops.model import ActiveStatus, BlockedStatus, RelationDataTypeError, WaitingStatus
+
+logger = logging.getLogger(__name__)
+REL_NAME = "gcs-credentials"
+BACKEND_NAME = "gcs"
+
+
+class GcsRequirerEvents(Object):
+    """Requirer-side helper which listens to lib events and sets status directly."""
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        relation_name: str = REL_NAME,
+    ):
+        super().__init__(charm, "gcs-requirer")
+        self.charm = charm
+        self.relation_name = relation_name
+        self.storage = StorageRequires(charm, relation_name, BACKEND_NAME, overrides={})
+        self.framework.observe(
+            self.storage.on.storage_connection_info_changed, self._on_conn_info_changed
+        )
+        self.framework.observe(
+            self.storage.on.storage_connection_info_gone, self._on_conn_info_gone
+        )
+        self.framework.observe(
+            self.storage.on[self.relation_name].relation_joined, self._on_relation_joined
+        )
+
+        self.framework.observe(self.charm.on.config_changed, self._on_config_changed)
+
+        self._last_sent_overrides: Dict[str, str] = {}
+
+    def _on_config_changed(self, _):
+        ov = self.overrides_from_config()
+        self.apply_overrides(ov)
+        self.refresh_status()
+
+    def _on_relation_joined(self, event):
+        ov = self.overrides_from_config()
+        self.apply_overrides(ov, relation_id=event.relation.id)
```

Comment:
Isn't this already handled by the library in the relation-joined event ? I remember [this line](https://github.com/canonical/object-storage-integrators/pull/47/files#diff-39f8c417c684c9fb3b40ba3cfdfa38838844d57b961f0c85719d06c85ccb1df0R434) that was supposed to do it automatically.
The thing is, with the way data interfaces was developped you were never supposed to handle regular events manually, just leave it to the library that would handle it for you. SO I'm wondering why the logic is duplicated here between the library and this listener.

The way this charm should be defined would be with:
```
self.storage = StorageRequires(charm, relation_name, BACKEND_NAME, overrides=self.overrides_from_config())
```
So that the library can handle it itself.